teamcity: failed tests on v1.1-beta.20170907: testrace/TestNodeLiveness

The following tests appear to have failed:

[#346836](https://teamcity.cockroachdb.com/viewLog.html?buildId=346836):

```
--- FAIL: testrace/TestNodeLiveness (0.590s)
node_liveness_test.go:96: heartbeat failed on epoch increment
------- Stdout: -------
I170907 19:53:27.947121 90706 storage/store.go:1193  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170907 19:53:27.947591 90706 gossip/gossip.go:311  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:57409" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W170907 19:53:28.021994 90706 gossip/gossip.go:1225  [n2] no incoming or outgoing connections
I170907 19:53:28.029689 90447 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:57409
I170907 19:53:28.046417 90706 gossip/gossip.go:311  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43121" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W170907 19:53:28.340840 90706 gossip/gossip.go:1225  [n3] no incoming or outgoing connections
I170907 19:53:28.343050 90871 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:57409
I170907 19:53:28.368261 90706 gossip/gossip.go:311  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:50139" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I170907 19:53:28.456503 90708 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:57409->127.0.0.1:51548: use of closed network connection
I170907 19:53:28.457237 90868 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:50139->127.0.0.1:56639: use of closed network connection
I170907 19:53:28.458077 90720 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:43121->127.0.0.1:48917: use of closed network connection

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/node_liveness.go
+++ b/pkg/storage/node_liveness.go
@@ -43,9 +43,9 @@ var (
 
 	errChangeDecommissioningFailed = errors.New("failed to change the decommissioning status")
 
-	// errSkippedHeartbeat is returned when a heartbeat request fails because
+	// ErrEpochIncremented is returned when a heartbeat request fails because
 	// the underlying liveness record has had its epoch incremented.
-	errSkippedHeartbeat = errors.New("heartbeat failed on epoch increment")
+	ErrEpochIncremented = errors.New("heartbeat failed on epoch increment")
 )
 
 type errRetryLiveness struct {
@@ -351,7 +351,7 @@ func (nl *NodeLiveness) StartHeartbeat(
 							log.Errorf(ctx, "unexpected error getting liveness: %v", err)
 						}
 						if err := nl.heartbeatInternal(ctx, liveness, incrementEpoch); err != nil {
-							if err == errSkippedHeartbeat {
+							if err == ErrEpochIncremented {
 								log.Infof(ctx, "%s; retrying", err)
 								continue
 							}
@@ -464,7 +464,7 @@ func (nl *NodeLiveness) heartbeatInternal(
 			return errNodeAlreadyLive
 		}
 		// Otherwise, return error.
-		return errSkippedHeartbeat
+		return ErrEpochIncremented
 	}); err != nil {
 		if err == errNodeAlreadyLive {
 			nl.metrics.HeartbeatSuccesses.Inc(1)

--- a/pkg/storage/node_liveness_test.go
+++ b/pkg/storage/node_liveness_test.go
@@ -29,6 +29,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/storagebase"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 )
 
@@ -92,7 +93,16 @@ func TestNodeLiveness(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
-		if err := nl.Heartbeat(context.Background(), l); err != nil {
+		for {
+			err := nl.Heartbeat(context.Background(), l)
+			if err == nil {
+				break
+			}
+			if err == storage.ErrEpochIncremented {
+				log.Warningf(context.Background(), "retrying after %s", err)
+				continue
+			}
+
 			t.Fatal(err)
 		}
 	}

